OpenRA.Mods.Common/Projectiles/NukeLaunch.cs

```diff
@@ -116,23 +125,33 @@ public void Tick(World world)
 				trailTicks = trailInterval;
 			}
 
+			if (isDescending && preDetonates && pos.Z <= detonationAltitude.Length)
```

Comment:
These two cases can be combined, and the `preDetonates` flag removed:
```csharp
if (ticks == impactDelay || (isDescending && pos.Z <= detonationAltitude.Length))
	Explode(world, ticks == impactDelay || removeOnDetonation);
```

Reply:
This also doesn't seem to be accounting for terrain height in TS - cluster missiles launched on high ground impact the ground then send clusters horizontally.

Reply:
Should be DistanceAboveTerrain instead of pos.Z, I guess.